File: app/src/main/java/com/sumant/todonotesapp/DetailActivity.java
package com.sumant.todonotesapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class DetailActivity extends AppCompatActivity {

    TextView titleTextView , descriptionTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        bindViews();
        getIntentData();
    }

    private void getIntentData() {
        titleTextView.setText( getIntent().getStringExtra(AppConstrain.TITLE) );
        descriptionTextView.setText( getIntent().getStringExtra(AppConstrain.DESCRIPTION) );
    }

    private void bindViews() {
        titleTextView = findViewById(R.id.titleDetailTextView);
        descriptionTextView = findViewById(R.id.descriptionDetailTextView);
    }
}

File: app/src/main/java/com/sumant/todonotesapp/DetailActivity.kt
package com.sumant.todonotesapp

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class DetailActivity : AppCompatActivity() {
    var titleTextView: TextView? = null
    var descriptionTextView: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        bindViews()
        intentData
    }

    private val intentData: Unit
        private get() {
            titleTextView!!.text = intent.getStringExtra(AppConstrain.TITLE)
            descriptionTextView!!.text = intent.getStringExtra(AppConstrain.DESCRIPTION)
        }

    private fun bindViews() {
        titleTextView = findViewById(R.id.titleDetailTextView)
        descriptionTextView = findViewById(R.id.descriptionDetailTextView)
    }
}